contracts/src/periphery/EigenDACertVerifierV1.sol

```diff
@@ -0,0 +1,152 @@
+
+    /// @notice The EigenDAThresholdRegistry contract address
+    IEigenDAThresholdRegistry public immutable eigenDAThresholdRegistryV1;
+
+    /// @notice The EigenDABatchMetadataStorage contract address
+    /// @dev On L1 this contract is the EigenDA Service Manager contract
+    IEigenDABatchMetadataStorage public immutable eigenDABatchMetadataStorageV1;
+
+    /**
+     * @notice Constructor for the EigenDA V1 certificate verifier
+     * @param _eigenDAThresholdRegistryV1 The address of the EigenDAThresholdRegistry contract
+     * @param _eigenDABatchMetadataStorageV1 The address of the EigenDABatchMetadataStorage contract
+     */
+    constructor(
+        IEigenDAThresholdRegistry _eigenDAThresholdRegistryV1,
+        IEigenDABatchMetadataStorage _eigenDABatchMetadataStorageV1
+    ) {
+        eigenDAThresholdRegistryV1 = _eigenDAThresholdRegistryV1;
+        eigenDABatchMetadataStorageV1 = _eigenDABatchMetadataStorageV1;
+    }
+
+    /**
+     * @notice Verifies a the blob cert is valid for the required quorums
+     * @param blobHeader Pointer to the blob header in calldata
+     * @param blobVerificationProof Pointer to the blob cert verification proof in calldata
+     */
+    function verifyDACertV1(BlobHeader calldata blobHeader, BlobVerificationProof calldata blobVerificationProof)
+        public
+        view
+    {
+        (CertLib.ErrorCode err, bytes memory errParams) = CertLib.verifyDACert(
+            _quorumConfirmationThresholdPercentages(),
+            _storedBatchMetadataHash(blobVerificationProof),
+            blobHeader,
+            blobVerificationProof,
+            _quorumNumbersRequired()
+        );
+        CertLib.revertOnError(err, errParams);
+    }
+
+    /**
+     * @notice Verifies a batch of blob certs for the required quorums
+     * @param blobHeaders Pointer to array of blob headers in calldata
+     * @param blobVerificationProofs Pointer to array of blob cert verification proofs in calldata
+     */
+    function verifyDACertsV1(BlobHeader[] calldata blobHeaders, BlobVerificationProof[] calldata blobVerificationProofs)
+        external
+        view
+    {
+        // Check length match
+        if (blobHeaders.length != blobVerificationProofs.length) {
+            revert LengthMismatch();
+        }
+
+        // Verify each blob
+        for (uint256 i; i < blobHeaders.length; ++i) {
+            verifyDACertV1(blobHeaders[i], blobVerificationProofs[i]);
+        }
+    }
+
+    /**
+     * @notice Verifies a blob cert and returns result without reverting
+     * @param blobHeader Pointer to the blob header in calldata
+     * @param blobVerificationProof Pointer to the blob cert verification proof in calldata
+     * @return success True if verification succeeded
+     */
+    function verifyDACertV1ForZkProof(
+        BlobHeader calldata blobHeader,
+        BlobVerificationProof calldata blobVerificationProof
+    ) external view returns (bool success) {
```

Comment:
technically we'd use this for fraud proofs too so the suffix `ZkProof` may need to be updated to something else

Reply:
Agreed it's not a great name, but it is consistent with the naming for the V2 Cert verifier which is taken from master.

Reply:
Also think we should change both. Made the same comment here https://github.com/Layr-Labs/eigenda/pull/1472#discussion_r2048232747

Reply:
iiuc the interface will have to change given that we plan to change the verification functions to also take in a structured EigenDA cert type?

Reply:
@ethenotethan I should mention that I discovered there are obstacles to reordering fields. The problem is that the batch header hash calculation depends on the field order in a batch, and so any reordering of these fields is a VERY breaking change. I think this should be done in another PR with a full discussion of implications.

Reply:
https://github.com/Layr-Labs/eigenda/pull/1477/commits/33e719bddee8e5b701c314100d1ceb1f1ca33faf